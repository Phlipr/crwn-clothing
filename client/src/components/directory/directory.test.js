import React from "react";
import { shallow } from "enzyme";

import Directory from "./directory.component";
import { useSelector } from "react-redux";
import { HashRouter } from "react-router-dom";

const mockState = {
    directory: {
        sections: [
            {
                title: "hats",
                id: 1
            },
            {
                title: "sneakers",
                id: 2
            },
            {
                title: "jackets",
                id: 3
            }
        ]
    }
}

jest.mock("react-redux", () => ({
    useSelector: jest.fn()
}));

describe("Directory component:", () => {
    it("should render the Directory component", () => {
        useSelector.mockImplementation(selector => selector(mockState));
        expect.assertions(1);
        expect(shallow(<Directory />)).toMatchSnapshot();
    });
});