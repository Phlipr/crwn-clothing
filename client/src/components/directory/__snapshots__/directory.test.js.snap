// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component: should render the Directory component 1`] = `ShallowWrapper {}`;
